src/environment.ts

```diff
@@ -192,7 +192,9 @@ export class SecureEnvironment {
 
     // membrane operations
     getSecureValue(raw: RawValue): SecureValue {
-        if (isProxyTarget(raw)) {
+        if (isArray(raw)) {
```

Comment:
the main bug here was that `isProxyTarget` returns true for arrays, since they are `typeof` object, which means we ended up creating a proxy for them.